spec/chartmogul/subscription_spec.rb

```diff
@@ -145,4 +145,36 @@
       data_source.destroy!
     end
   end
+
+  describe '#update_cancellation_dates', uses_api: true, vcr: true do
+    let(:external_id) { 'test_cus_ext_id' }
+    let(:customer) { ChartMogul::Customer.all(external_id: external_id).first }
+    let(:subscription) { customer.subscriptions.first }
+
+    subject { subscription.update_cancellation_dates(dates) }
+
+    describe 'when array is empty' do
+      let(:dates) { [] }
+
+      it 'makes an API call and removes the cancellation dates' do
+        expect(subject.cancellation_dates).to eq []
+      end
+    end
+
+    describe 'when array includes invalid entries' do
+      let(:dates) { ['invalid_entry'] }
+
+      it 'raises an exception' do
+        expect { subject }. to raise_error ArgumentError
+      end
+    end
+
+    describe 'when array includes valid entries' do
+      let(:dates) { ['2000-01-01'] }
```

Comment:
Please add a test case for sending an array of Time objects, which ppl will probably do.

Reply:
Sure